Add warning notice on GitHub Jobs Deprecation

rlaphoenix opened the pull request:
### Why:

Closes https://github.com/github/docs/issues/6574

### What's being changed:

It adds a warning box with information relating to the deprecation and eventual shutdown of jobs.github.com, linking to the blog post. All wording and typesetting are from the jobs.github.com deprecation warning banner.

### Check off the following:

- [X] I have reviewed my changes in staging (look for the **deploy-to-heroku** link in your pull request, then click **View deployment**).
- [X] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/CONTRIBUTING.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

<!-- Description of the writer impact here -->

rlaphoenix committed:
Add warning notice on GitHub Jobs Deprecation

More information is available at the Blog: https://github.blog/changelog/2021-04-19-deprecation-notice-github-jobs-site/

It's probably ideal to remove this entire docs page after the checkbox is removed (if it gets removed) or updated to state it is shut down rather than deprecated once the full shutdown has taken place.

sarahs commented:
👋 We did some content restructuring that resulted in a conflict with your PR. It looks like we may not have access to check out your fork. Let us know if you need a hand resolving the conflict!

Commit:
Resolve Conflict for location of the markdown file

Commit:
Resolve file change conflict of markdown file

rlaphoenix committed:
Merge branch 'main' into patch-1

Commit:
Also add a deprecation warning to about-github-jobs

rlaphoenix commented:
I've fixed the conflict, and I've also gone ahead and added the same warning message to https://docs.github.com/en/github/finding-talent-with-github-jobs/about-github-jobs

janiceilene commented:
@rlaphoenix Thanks so much for opening a PR! I'll get this triaged for review ⚡

jmarlena committed:
Merge branch 'main' into patch-1

jmarlena commented on content/github/finding-talent-with-github-jobs/about-github-jobs.md:
```diff
@@ -10,6 +10,13 @@ topics:
 
+{% warning %}
+
+GitHub Jobs is deprecated! New jobs will not be posted from May 19, 2021. It will shut down entirely on August 19, 2021.
```
```suggestion
Deprecation note: GitHub Jobs is now deprecated. The last date to post a job was May 19, 2021. The GitHub Jobs site will shut down entirely on August 19, 2021. For more information, see the [GitHub blog post](https://github.blog/changelog/2021-04-19-deprecation-notice-github-jobs-site/).
```

jmarlena commented on content/github/finding-talent-with-github-jobs/about-github-jobs.md:
```diff
@@ -10,6 +10,13 @@ topics:
+{% warning %}
+
+GitHub Jobs is deprecated! New jobs will not be posted from May 19, 2021. It will shut down entirely on August 19, 2021.
+Read more in the [GitHub blog post](https://github.blog/changelog/2021-04-19-deprecation-notice-github-jobs-site/).
```
```suggestion

```

jmarlena committed:
Small style updates

jmarlena commented on content/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/what-does-the-available-for-hire-checkbox-do.md:
```diff
@@ -9,6 +9,14 @@ versions:
+
+{% warning %}
+
+GitHub Jobs is deprecated! New jobs will not be posted from May 19, 2021. It will shut down entirely on August 19, 2021.
```
```suggestion
Deprecation note: GitHub Jobs is now deprecated. The last date to post a job was May 19, 2021. The GitHub Jobs site will shut down entirely on August 19, 2021. For more information, see the [GitHub blog post](https://github.blog/changelog/2021-04-19-deprecation-notice-github-jobs-site/).
```

jmarlena commented on content/github/setting-up-and-managing-your-github-user-account/managing-user-account-settings/what-does-the-available-for-hire-checkbox-do.md:
```diff
@@ -9,6 +9,14 @@ versions:
+{% warning %}
+
+GitHub Jobs is deprecated! New jobs will not be posted from May 19, 2021. It will shut down entirely on August 19, 2021.
+Read more in the [GitHub blog post](https://github.blog/changelog/2021-04-19-deprecation-notice-github-jobs-site/).
```
```suggestion

```

jmarlena committed:
Last batch of style changes

jmarlena reviewed (approved):
Thanks for this thoughtful update!! I made some small style tweaks. 

This PR looks good to merge once tests pass. 🚢
